monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -3719,6 +3724,33 @@ abstract class Observable[+A] extends Serializable { self =>
   final def filter(p: A => Boolean): Observable[A] =
     self.liftByOperator(new FilterOperator(p))
 
+  /** Version of [[filter]] that can work with a predicate expressed by
+    * a [[monix.eval.Task]].
+    *
+    * @see [[filterEvalF]] for a version that works with a generic
+    *      `F[_]` (e.g. `cats.effect.IO`, Scala's `Future`),
+    *      powered by [[monix.eval.TaskLike]]
+    */
+  final def filterEval(p: A => Task[Boolean]): Observable[A] =
+    self
+      .mapEval(a ⇒ p(a).map((a, _)))
+      .filter(_._2)
+      .map(_._1)
+
+  /** Version of [[filterEval]] that can work with generic
+    * `F[_]` tasks, anything that's supported via [[monix.eval.TaskLike]]
+    * conversions.
+    *
+    * So you can work among others with:
+    *
+    *  - `cats.effect.IO`
+    *  - `monix.eval.Coeval`
+    *  - `scala.concurrent.Future`
+    *  - ...
+    */
+  final def filterEvalF[F[_]](p: A => F[Boolean])(implicit F: TaskLike[F]): Observable[A] =
+    filterEval(a => Task.from(p(a))(F))
```

Comment:
This too needs a test.

Reply:
https://github.com/monix/monix/pull/760/commits/839b500082d314ea6a3b732afff6c5ddbe8b4c95